src/modules/TONContractsModule.js

```diff
@@ -974,20 +1049,30 @@ export default class TONContractsModule extends TONModule implements TONContract
             });
         } catch (error) {
             this.config.log('[waitForTransaction]', 'FAILED', error);
-            if (TONClientError.isMessageExpired(error) ||
-                TONClientError.isClientError(error, TONClientError.code.TRANSACTION_WAIT_TIMEOUT)) {
-                throw await this.resolveDetailedError(
+            if (TONClientError.isMessageExpired(error)
+                || TONClientError.isClientError(error, TONErrorCode.TRANSACTION_WAIT_TIMEOUT)) {
+                const detailedError = await this.resolveDetailedError(
                     error,
                     message.messageBodyBase64,
                     Date.now(),
                     message.address,
                 );
+                (detailedError: any).data = {
```

Comment:
Better take all needed fields from `error.data` explicitly, not the whole object